guides/micronaut-data-one-to-many/java/src/main/java/example/micronaut/PhoneEntity.java

```diff
@@ -0,0 +1,24 @@
+package example.micronaut;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.data.annotation.GeneratedValue;
+import io.micronaut.data.annotation.Id;
+import io.micronaut.data.annotation.MappedEntity;
+import io.micronaut.data.annotation.Relation;
+
+@MappedEntity("phone") // <1>
+public record PhoneEntity(
+        @Id // <2>
+        @GeneratedValue  // <3>
+        @Nullable
```

Comment:
remove `Nullable` here as it can add confusion since from a database perspective the ID is never nullable

Reply:
https://micronaut-projects.github.io/micronaut-data/4.9.3/guide/#javaRecords

> Since records are immutable constructor arguments that are generated values need to be marked as @Nullable and you should pass null for those arguments

@dstepanov does this no longer apply?

Reply:
@graemerocher I added a callout

Reply:
I don't know why it is there; you must pass null for Micronaut / DB to generate it. There might be a problem if you have two generated values, and one is being set by recreating the record, which might trigger an error because one of the values is still null.

Reply:
since you must pass null, @graemerocher I think it makes more sense to keep the `@Nullable` annotation. Don't you think?